pkg/reconciler/source/config_watcher.go

```diff
@@ -20,143 +20,232 @@ import (
 	"context"
 
 	"go.uber.org/zap"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/pkg/logging"
 	"knative.dev/pkg/configmap"
-	pkgLogging "knative.dev/pkg/logging"
+	pkglogging "knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
 	tracingconfig "knative.dev/pkg/tracing/config"
+)
 
-	"knative.dev/eventing/pkg/logging"
+const (
+	EnvLoggingCfg = "K_LOGGING_CONFIG"
+	EnvMetricsCfg = "K_METRICS_CONFIG"
+	EnvTracingCfg = "K_TRACING_CONFIG"
 )
 
+type EnvVarsGenerator interface {
```

Comment:
That was my first though as well, but then I realized the current implementation only exposes `ToEnvVars`, so the intention behind my interface is not to wrap a ConfigWatcher, but rather to provide that one specific function (thing `Writer`, `ReadCloser`, etc. in the stdlib). This pattern of tiny interfaces is used a lot in Go.
If I implement a  `ConfigWatcher` interface, I expect it to at least implement methods like `Watch()`, `AddConfig()`, etc. Maybe we need to think twice about the naming.

Reply:
Given that: https://github.com/knative/eventing/pull/3109#discussion_r422819602 I think `EnvVarsGenerator` is not correct... Maybe `ConfigAccessor`? `SourceConfigs`? I don't know, unless you have better ideas, let's stick with `ConfigWatcher`

Reply:
I like `ConfigAccessor` 👍

Reply:
Ok please change the impl name too :wink:

Reply:
To what, sorry?

Reply:
`configAccessorImpl`

Reply:
I have 2 good reasons to respectfully disagree:
* following that logic, you'll end up with `configAccessorImpl1`, `configAccessorImpl2` in case you want to have multiple implementations 🚫 
* following that same logic, you'll end up with `configAccessorImplOtherInterfaceImpl` in case you implement multiple interfaces 🚫 

Go types can implement multiple interfaces so they usually describe what the _type_ is, not the interface(s) they implement.

Reply:
> following that logic, you'll end up with configAccessorImpl1, configAccessorImpl2 in case you want to have multiple implementations 

Why would you need multiple implementations in this module?

> following that same logic, you'll end up with configAccessorImplOtherInterfaceImpl in case you implement multiple interfaces 

But the implementation in this module implements only the `ConfigAccessor` interface

Reply:
Ah ok i see now that there is another blank implementation for tests. Well, name it as you want, as soon as is private :smile: